### PR TITLE
[NOT FOR MERGE] Clevo nv41mz/upstream

### PR DESCRIPTION
**Pull request by miczyg1**

### CODE REVIEW

File: payloads/external/tianocore/Makefile [hunk 1]
```diff
@@ -10,7 +10,12 @@ BUILD_STR = -a IA32 -a X64 -t COREBOOT
 ifeq ($(CONFIG_TIANOCORE_COREBOOTPAYLOAD),y)
 BUILD_STR += -p CorebootPayloadPkg/CorebootPayloadPkgIa32X64.dsc
 else
+ifeq ($(CONFIG_TIANOCORE_UPSTREAM)$(CONFIG_TIANOCORE_UEFIPAYLOAD),y)
 BUILD_STR += -p UefiPayloadPkg/UefiPayloadPkg.dsc
+else
+# fallback for the old mixed IA32 X64 build
+BUILD_STR += -p UefiPayloadPkg/UefiPayloadPkgIa32X64.dsc
```
Comment (mkopec) [inline payloads/external/tianocore/Makefile:17]: TODO: copy this file to `UefiPayloadPkg.dsc`

Comment (mkopec) [reply payloads/external/tianocore/Makefile]: Done in https://github.com/Dasharo/edk2/pull/7. Line removed here

File: payloads/external/tianocore/Kconfig [hunk 2]
```diff
@@ -146,4 +146,15 @@ config TIANOCORE_USE_8254_TIMER
 
 endif
 
+if TIANOCORE_CUSTOM
```
Comment (mkopec) [inline payloads/external/tianocore/Kconfig:149]:
```suggestion
if !TIANOCORE_CUSTOM
```

File: configs/config.clevo_nv41mz [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+CONFIG_SMMSTORE_SIZE=0x40000
+# CONFIG_TPM_PPI is not set
+CONFIG_CBFS_MCACHE_RW_PERCENTAGE=50
+CONFIG_VBOOT_KEYBLOCK_VERSION=1
+CONFIG_VBOOT_KEYBLOCK_PREAMBLE_FLAGS=0x0
+CONFIG_DEFAULT_CONSOLE_LOGLEVEL_0=y
+CONFIG_POST_IO_PORT=0x80
+CONFIG_PAYLOAD_TIANOCORE=y
+CONFIG_PAYLOAD_FILE="uefipayload.elf"
+CONFIG_TIANOCORE_CUSTOM=y
```
Comment (mkopec) [inline configs/config.clevo_nv41mz:27]:
```suggestion
```

File: payloads/external/tianocore/Makefile [hunk 4]
```diff
@@ -54,6 +54,12 @@ endif
 
 bootloader = $(word 8,$(subst /, ,$(BUILD_STR)))
 
+ifeq ($(CONFIG_TIANOCORE_CUSTOM),y)
```
Comment (mkopec) [inline payloads/external/tianocore/Makefile:57]:
```suggestion
ifneq ($(CONFIG_TIANOCORE_CUSTOM),y)
```

**Comment by mkopec** [comment]
Fixed TianoCore build problems and rebased on top of upstream coreboot tag v4.16.

**Comment by mkopec** [comment]
Renamed the mainboards to `nv4x` and `ns5x`. Added the `*_BASE` variants to be able to use the code in other mainboards.

**Comment by pietrushnic** [comment]
@miczyg1 @mkopec why this MR is open? We should move that stuff to upstream.

**Comment by mkopec** [comment]
@pietrushnic We moved this upstream, forgot to close this PR: Upstream URL: https://review.coreboot.org/c/coreboot/+/62498

Closing.

**Pull request closed without merging**